<?xml version="1.0" encoding="utf-8"?>
<launch>
    <arg name="veh" default="$(env VEHICLE_NAME)" />
    <arg name="config" default="baseline" />
    <arg name="param_file_name" default="default" />
    
    <!-- Camera parameters -->
    <arg name="framerate" default="30" doc="Camera framerate" />
    <arg name="res_w" default="640" doc="Camera resolution width" />
    <arg name="res_h" default="480" doc="Camera resolution height" />
    <arg name="exposure_mode" default="auto" doc="Camera exposure mode" />
    <arg name="camera_type" default="auto" doc="Camera type: auto, pi, usb, jetson_csi, or custom" />
    <arg name="flip_method" default="0" doc="Jetson nvvidconv flip-method (0..7)" />
    <arg name="sensor_mode" default="" doc="Jetson nvarguscamerasrc sensor-mode (empty for default)" />
    
    <group ns="$(arg veh)">
        <!-- Auto-detect and launch appropriate camera node -->
        <node name="camera_node" 
              pkg="camera" 
              type="camera_driver_node.py" 
              output="screen">
            <param name="framerate" value="$(arg framerate)" />
            <param name="res_w" value="$(arg res_w)" />
            <param name="res_h" value="$(arg res_h)" />
            <param name="exposure_mode" value="$(arg exposure_mode)" />
            <param name="camera_type" value="$(arg camera_type)" />
            <param name="veh_name" value="$(arg veh)" />
            <param name="flip_method" value="$(arg flip_method)" />
            <param name="sensor_mode" value="$(arg sensor_mode)" />
        </node>
    </group>
</launch>